api/src/routes/ranks.ts

```diff
@@ -0,0 +1,287 @@
+ *       500:
+ *         description: Internal Server Error
+ */
+
+router.get("/user", async (req: Request, res: Response) => {
+  try {
+    const { userId, gameId } = req.query;
+
+    const result = await pool.query(
+      "SELECT * FROM user_ranks WHERE user_id = $1 AND game_id = $2",
+      [userId, gameId]
+    );
+    res.json(result.rows);
+  } catch (error) {
+    logger.error(error);
+    res.status(500).send("Internal Server Error");
+  }
+});
+
+/**
+ * @swagger
+ * /ranks/user/{userId}:
+ *   get:
+ *     summary: Get rank by user id
+ *     tags: [Ranks]
+ *     description: Retrieve all ranks for a specific user from the database.
+ *     parameters:
+ *       - in: path
+ *         name: userId
+ *         required: true
+ *         description: ID of the user to retrieve ranks for
+ *         schema:
+ *           type: integer
+ *       - in: query
+ *         name: page
+ *         description: Page number for pagination
+ *         schema:
+ *           type: integer
+ *         default: 1
+ *       - in: query
+ *         name: pageSize
+ *         description: Number of logs per page
+ *         schema:
+ *           type: integer
+ *         default: 10
+ *     responses:
+ *       200:
+ *         description: Successful response
+ *       500:
+ *         description: Internal Server Error
+ */
+
+router.get("/user/:userId", async (req: Request, res: Response) => {
+  try {
+    const { userId } = req.params;
+    const { page = 1, pageSize = 10 } = req.query;
+
+    if (!paginationIsValid(page as number, pageSize as number)) {
+      return res
+        .status(400)
+        .json({ error: "Invalid page or pageSize parameters" });
+    }
+
+    // Calculate the offset for pagination
+    const offset = (Number(page) - 1) * Number(pageSize);
+    const result = await pool.query(
+      "SELECT * FROM user_ranks WHERE user_id = $1 ORDER BY rank_date DESC LIMIT $2 OFFSET $3",
+      [userId, Number(pageSize), offset]
+    );
+    res.json(result.rows);
+  } catch (error) {
+    logger.error(error);
+    res.status(500).send("Internal Server Error");
+  }
+});
+
+/**
+ * @swagger
+ * /ranks/average:
+ *   get:
+ *     summary: Get an average of the ranks of a user on a game
+ *     tags: [Ranks]
+ *     description: Retrieve the average rank of a user on a specific game from the database.
+ *     parameters:
+ *       - in: query
+ *         name: userId
+ *         required: true
+ *         description: ID of the user to retrieve ranks for
+ *         schema:
+ *           type: integer
+ *       - in: query
+ *         name: gameId
+ *         required: true
+ *         description: ID of the game to retrieve ranks for
+ *         schema:
+ *           type: integer
+ *     responses:
+ *       200:
+ *         description: Successful response
+ *       500:
+ *         description: Internal Server Error
+ */
+
+// TODO: Add a since/until parameter.
+router.get("/average", async (req: Request, res: Response) => {
```

Comment:
Same here, thinking this would be `GET /users/{user_id}/games/{game_id}/average-rank`. I know it is long an chunky but looking at restful API naming (https://restfulapi.net/resource-naming/). Please raise if you disagree!

Reply:
I agree it should be a user endpoint, but based on a skim over what you sent- I would argue it should be: `/users/{userId}/ranks/average`

I've only skimmed so I'll read properly but this is my initial reaction.

Reply:
From what I understand from that page, and my own knowledge. Query params should only be used for sorting, filtering, and pagination capabilities. Where as game ID is none of these and a table (noun) itself but your logic asks for it a as query parameter!

Reply:
Discussed in person: will use: `/users/{userId}/games/{gameId}/ranks/average`